sunpy/map/mapbase.py

```diff
@@ -178,7 +178,7 @@ def __init_subclass__(cls, **kwargs):
         if hasattr(cls, 'is_datasource_for'):
             cls._registry[cls] = cls.is_datasource_for
 
-    def __init__(self, data, header, plot_settings=None, **kwargs):
+    def __init__(self, data, header, plot_settings=None, mask=None, **kwargs):
```

Comment:
Was mask missing and not handled by kwargs?

Reply:
The parent class `NDData` doesn't support kwargs, so I made `mask` an  explicit arg and removed kwargs from super().
https://github.com/sunpy/sunpy/blob/0f7946d90dae9eeb6ae8e81913a32e0745ee7b44/sunpy/map/mapbase.py#L196

I think there's a test, but since `mask` is an explicit arg in `nddata` it didn't raise any exceptions.

Reply:
So the test we have didn't work?

Reply:
the test passes.
But an exception is raised if we pass any kwarg which is not in `nddata`.
for example,
```python
sunpy.map.Map(fname, filetype='fits')
```
would raise an exception.

Reply:
Oh I see.